src/ExpressionEvaluator/CSharp/Test/ResultProvider/DebuggerBrowsableAttributeTests.cs

```diff
@@ -114,6 +114,33 @@ class C
                 EvalResult("P6", "6", "object {int}", "((B)c.o).P6", DkmEvaluationResultFlags.ReadOnly));
         }
 
+        [Fact]
+        public void HiddenMembers()
+        {
+            var source =
+@"using System.Diagnostics;
+abstract class A
+{
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+    public object P1;
+}
+class B : A
+{
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+    new public object P1 => base.P1;
```

Comment:
Does this test case hit the issue of duplicate attributes in the test infrastructure? Isn't `B.P1` a distinct member from `A.P1`?

Do we hit a similar issue with a virtual property that is overridden in the derived type? Please add a test for that case as well.

Reply:
Consider also testing cases with `new` and `override` properties in the derived class where `[DebuggerBrowsable(Never)]` is set on only the base class member or only in the derived class.

Reply:
This is covered in the existing Never_OverridesAndImplements test

---
In reply to: [251188530](https://github.com/dotnet/roslyn/pull/26049#discussion_r251188530) [](ancestors = 251188530)

Reply:
Added

---
In reply to: [251188340](https://github.com/dotnet/roslyn/pull/26049#discussion_r251188340) [](ancestors = 251188340)

Reply:
It looks like `Never_OverridesAndImplements` tests the `override` cases but not the `new` cases.

Will we display any members in this case?
```C#
class A
{
    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
    public object P1;
    internal object P2 => 0;
}
class B : A
{
    new public object P1 => 1;
    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
    internal new object P2 => 2;
}
```

---
In reply to: [251990380](https://github.com/dotnet/roslyn/pull/26049#discussion_r251990380) [](ancestors = 251990380,251188530)

Reply:
Thank you!
Yes, we will display P2 with the following cast ((A)this).P3. I've added this to the test.

---
In reply to: [252776364](https://github.com/dotnet/roslyn/pull/26049#discussion_r252776364) [](ancestors = 252776364,251990380,251188530)

Reply:
Thanks. I think the EE should include `B.P2` in the expansion of `B` because `P2` is a `new` property with no attribute. That behavior is not the result of this PR though, and doesn't need to be fixed here. Please consider logging a bug for that issue.